Editor/AGS.Editor/Resources/agsdefns.sh

```diff
@@ -406,6 +406,19 @@ enum SortStyle
 };
 #endif
 
+#ifdef SCRIPT_API_v351
+enum LogLevel
+{
+	eLogAlert = 1,
+	eLogFatal = 2,
+	eLogError = 3,
+	eLogWarn = 4,
+	eLogInfo = 5,
+	eLogDebug = 6
+};
+#endif

```

Comment:
i just tried to use this on master but it doesn't seem to be there ??

Reply:
It's on ags4 branch

Reply:
well, it's using api level 3.5.1, and 3.6.0 supercedes 3.5.1. it being in ags4 only is clearly a mistake.

Reply:
It's using api 3.5.1 in this pr. Currently in ags4 it refers to another api level: https://github.com/adventuregamestudio/ags/blob/ags4/Editor/AGS.Editor/Resources/agsdefns.sh#L400

Originally there was a branch for 3.5.1 that was discontinued.

Reply:
This is all happening because the plans for ags3/ags4 development are changing all the time.

Currently I'm actually considering to copy some of these back into 3.5.* or 3.6 because ags4 has some long-term goals, and particular API requires trivial changes to the engine, and may be useful in ags3 too.

Reply:
if we are going to add something back then we better have this well defined to tell the Scumm VM people about it, since so far ags3 is the branch that is safe to run in scummvm.

Reply:
50% of the comments above were about whether this makes it into ags3 or ags4 and the answer was always "ags3"

if this is ags4 only, it is useless for me

Reply:
> if we are going to add something back then we better have this well defined to tell the Scumm VM people about it, since so far ags3 is the branch that is safe to run in scummvm.

It's not like I want to add this to already released version. If the scummvm team's goal is to sync with our 3.* branch, they are likely keep a close eye on our releases and changes.

Reply:
> if this is ags4 only, it is useless for me

Very well, I will see to port it to current master (3.6.0). Probably will take me few days to do everything carefully.
There was a number of functions, including this log, that were relatively simple and do not require big changes to the engine.

Reply:
don't worry about scummvm, we discussed in #640 that we need a script function that can log to stdout for *debugging* purposes so its unlikely this is going to be used by game authors.

> Edit: btw System.Log is not a printf if that's what you were thinking...

what you mean by that? it's the debug printf function we've all been waiting for 1.5 years since #640 was opened, until it was finally merged into ags3 and then magically disappeared from ags3 just to re-appear in ags4 branch, and hurray, today when i finally got around to hacking on ags script and required this function for debugging, i noticed that it's no longer there.

Reply:
The System.Log prints to whatever engine is printing to. If engine prints to stdout, then the System.Log will print to stdout.

Reply:
I needed it for my tap module, but if I am not mistaken it doesn't print the text purely, it adds a prefix, which breaks for me, I needed a clean print so that the CI could pickup the test results.

Reply:
Well, I did not know that. I think there has been a separate idea of something like System.PrintStdOut().

Reply:
@rofl0r I ported System.Log back to master: 0c226592c679106e15f4ec3847ee0d57c2afae19

It is supposed to work as described in this PR (logging level is controlled by engine's log config).

sorry for the inconvenience.

Reply:
great, thanks! this is probably one of the most useful additions to ags in last 2 years.